_posts/2023-11-01-#28368.md

```diff
@@ -7,18 +7,73 @@ authors: [ismaelsadeeq]
+
+
+* This PR removes `CTxMempool` dependency on `CBlockPolicyEstimator` by making `CBlockPolicyEstimator` a client of `CValidationInterface`, instead of an internal component of `CTxMemPool`.
+
+  * The `CBlockPolicyEstimator` now updates asynchronously in response to `CValidationInterface` callback notifications.
+  * The PR added another `CValidationInterface` callback `MempoolTransactionsRemovedForConnectedBlock` which is fired in [`removeForBlock`](https://github.com/bitcoin/bitcoin/blob/ab2f531b785c3c17746ac9766a6db5b6d26677e8/src/txmempool.h#L487).
+  * The PR also modifies the `TransactionAddedToMempool` callback parameter from [`CTransactionRef`](https://github.com/bitcoin/bitcoin/blob/ab2f531b785c3c17746ac9766a6db5b6d26677e8/src/primitives/transaction.h#L421) to a new struct `NewMempoolTransactionInfo` which has `CTransactionRef` and other information the fee estimator needs such as the size, fees, and others.
+* There has been an attempt at this PR [#11775](https://github.com/bitcoin/bitcoin/pull/11775)
+
 ## Questions
+
+### Conceptual
 1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)? What was your review approach?
--->
 
+2. Why is it beneficial to remove `CTxMempool` dependency on `CBlockPolicyEstimator`?
+
+3. Are there any benefits of the `CBlockPolicyEstimator` being a member of `CTxMempool`? Are there downsides to removing it?
+
+4. The first attempt [#11775](https://github.com/bitcoin/bitcoin/pull/11775) split the `CValidationInterface` into `CValidationInterface` and `MempoolInterface`. What is the distinction between the two interfaces?
+
+5. What do you think is better or worse about the approach taken in this approach, versus the one taken in [#11775](https://github.com/bitcoin/bitcoin/pull/11775) split the `CValidationInterface`?
+
+6. Can you list the other subscribers to `CValidationInterface`?
+
+7. Why is implementing a `CValidationInterface` method equivalent to "subscribing to the event"?
+
+8. `BlockConnected` and `NewPoWValidBlock` are different callbacks. Are they both asynchronous?
+
+### Code Review
+
+{:start="10"}
+
+9. In [`4986edb`](https://github.com/bitcoin-core-review-club/bitcoin/commit/4986edb99f8aa73f72e87f3bdc09387c3e516197), why are we adding a new callback `MempoolTransactionsRemovedForConnectedBlock` instead of using `BlockConnected`?
+
+10. In [`1d116df`](https://github.com/bitcoin-core-review-club/bitcoin/commit/1d116df4c0e021c4c810450e3e5358f34d72940b), is `kernel/mempool_entry.h` the right place for `NewMempoolTransactionInfo`? What can be a better name for the struct?
```

Comment:
I don't think we'll learn very much from discussing what to name it, maybe talk about its members instead?
```suggestion
10. In [`1d116df`](https://github.com/bitcoin-core-review-club/bitcoin/commit/1d116df4c0e021c4c810450e3e5358f34d72940b), is `kernel/mempool_entry.h` the right place for `NewMempoolTransactionInfo`? What members are included in this struct, and why are they necessary?
```